File: Ruby/BasicAlgorithms/PrintOddsTo255.rb
module PrintOddsTo255

  def PrintOddsTo255.For
    for i in 1..255
      if i.modulo(2) == 1 
        print  "For to 255 odds value: ", i, "\n"
      end
    end
  end

  def PrintOddsTo255.DoWhile
    x = 0
    begin
      if x.modulo(2) == 1 
        print  "Do While to 255 odds value: ", x, "\n"
      end
      x += 1
    end while x <256
  end

  def PrintOddsTo255.DoUntil
    x = 0
    until x == 256 do
      if x.modulo(2) == 1 
        print  "Do Until to 255 odds value: ", x, "\n"
      end
      x += 1
    end
  end

end

File: JavaScript/BasicAlgorithms/PrintOddsTo255.js
/* Print all of the odd integers from 1 to 255.*/

export function For()
{
    for(var iterator = 0; iterator < 256; iterator++)
    {
        if(iterator % 2 === 1)
        {
            console.log(`For odds to 255  ${iterator}`);
        }
    }
}

export function ForIn()
{
    var arr = [...Array(256).keys()];
    for(var number in arr)
    {
        if(number % 2 === 1)
        {            
            console.log(`For In odds to 255  ${number}`);
        }
    }
}

export function ForOf()
{
    var arr = [...Array(256).keys()];
    for(var number of arr)
    {
        if(number % 2 === 1)
        {            
            console.log(`For Of odds to 255  ${number}`);
        }
    }
}

export function While()
{
    let iterator = 0;
    while(iterator < 256)
    {
        if(iterator % 2 === 1)
        {            
            console.log(`While to odds 255  ${iterator}`);
        }
        iterator++;
    }
}

export function DoWhile()
{
    let iterator = 0;
    do{
        if(iterator % 2 === 1)
        {            
            console.log(`Do While to odds 255  ${iterator}`);
        }
        iterator++;
    }
        
    while(iterator < 256)
}